multimodal/src/autogluon/multimodal/data/infer_types.py

```diff
@@ -239,13 +240,87 @@ def is_text_column(X: pd.Series) -> bool:
     return True
 
 
+def is_identifier_column(data: pd.Series, col_name: str, id_mappings: Dict[str, Dict]) -> bool:
+    """
+    Check if a column is one identifier column.
+
+    Parameters
+    ----------
+    data
+        One column of multimodal pd.DataFrame.
+    col_name
+        Name of the column.
+    id_mappings
+        Id-to-content mappings. The contents can be text, image, etc.
+        This is used when the dataframe contains the query/response indexes instead of their contents.
+
+    Returns
+    -------
+    Whether the column is an identifier column.
+    """
+    if not id_mappings or col_name not in id_mappings:
+        return False
+
+    sample_num = min(len(data), 500)
+    data = data.sample(n=sample_num, random_state=0).tolist()
+    failure_count = 0
+    for index in data:
+        try:
+            per_value = id_mappings[col_name][index]
+        except:
+            failure_count += 1
+
+    if failure_count == 0:
+        return True
+    elif 0 < failure_count < sample_num:
+        raise warnings.warn(
```

Comment:
seems that raise warnings will cause "exceptions must derive from BaseException": https://github.com/sonus21/error-tracker/issues/18

Reply:
Good catch! Thanks.